STYLE_GUIDE.md

```diff
@@ -1,3 +1,66 @@
 # Fortran stdlib Style Guide
 
-This document will describe the code style to use.
+> [A]utomate and codify as much as you possibly can while remembering that the human touch is still necessary to praise
+> and be kind to your contributors.
+> Let robots handle your project’s pedantry and humans handle your project’s empathy.
+
+-- @mikemcquaid, [Homebrew] project leader<sup>[1]</sup>
+
+[1]: https://mikemcquaid.com/2018/06/05/robot-pedantry-human-empathy/
+[Homebrew]: https://brew.sh
+
+## File naming conventions
+
+- Source files should contain at most one `program` or `module`/`submodule`.
+- The filename should match the program or module name and have the file extension `.f90` or `.F90` if preprocessing is required.
```

Comment:
As far as I understand, `.F90` for files to be preprocessed is only useful for the compiler to imply preprocessing. We can explicitly enable preprocessing with a flag (`-cpp`). Should we recommend just the `.f90` suffix?

Reply:
There is a *defacto* convention for using `.F90` and letting the compiler do the preprocessing. I would rather stick to this established convention. It indicates to the reader what to expect and works more smoothly with compilers and build tools like CMake.